Worksheet_D/oxo.py

```diff
@@ -0,0 +1,117 @@
+from __builtin__ import xrange
+
+
+class OxoBoard:
+    def __init__(self):
+        """ The initialiser. Initialise any fields you need here. """
+        self.oxoboard = [0, 0, 0, 0, 0, 0, 0, 0, 0]
+
+    def get_square(self, x, y):
+        """ Return 0, 1 or 2 depending on the contents of the specified square. """
+        Tile = x + y * 3
+        return self.oxoboard[Tile]
+
+
+    def set_square(self, x, y, mark):
+        """ If the specified square is currently empty (0), fill it with mark and return True.
+            If the square is not empty, leave it as-is and return False. """
+        Tile = x + y * 3
+        if self.oxoboard[Tile] == 0:
+            self.oxoboard[Tile] = mark
+            return True
+        else:
+            return False
+
+    def is_board_full(self):
+        """ If there are still empty squares on the board, return False.
+            If there are no empty squares, return True. """
+        for i in xrange (9):
+            if self.oxoboard[i] == 0:
+                return False
+        return True
+
+    def get_winner(self):
+        """ If a player has three in a row, return 1 or 2 depending on which player.
+            Otherwise, return 0. """
+        if self.oxoboard[0] == self.oxoboard[1] == self.oxoboard[2] != 0:
+            return self.oxoboard[0]
+        if self.oxoboard[3] == self.oxoboard[4] == self.oxoboard[5] != 0:
+            return self.oxoboard[3]
+        if self.oxoboard[6] == self.oxoboard[7] == self.oxoboard[8] != 0:

```

Comment:
You could reduce the repetition in the code by checking the three rows and three columns in a loop